.github/workflows/build-hyrise.yml

```diff
@@ -0,0 +1,80 @@
+# How to test this?
+#
+# In the root of the repository, run:
+#   nix-shell -p act --run "act"
+# Act is a software that allows you to simulate GitHub Actions CIs locally.
+# It works best in container environments that are not self-hosted.
+# Docker is required to run this as docker-credential-desktop is required,
+# which Nix does not deliver.
+
+name: Nix-Build of Hyrise
+on: push
+
+jobs:
+  nix-gcc-debug:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@main
+      - name: Nix - Hyrise Debug Clang Build
+        run: |
+          nix-shell --pure --run "\
+            mkdir nix-cmake-build && cd nix-cmake-build && \
+            cmake -DCMAKE_BUILD_TYPE=Debug .. && \
+            make -W hyriseServer
+          "
+
+  nix-gcc-release:
```

Comment:
Remove GCC parts?

Reply:
Yes, I wasn't sure whether we wanted to advance that path.